<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>COSMOS-tutorial • cosmosR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="COSMOS-tutorial">
<meta property="og:description" content="cosmosR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-PV71G2Q3SF"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-PV71G2Q3SF');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cosmosR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/tutorial.html">COSMOS-tutorial</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/saezlab/COSMOSR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/saezlab">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../www.saezlab.org">
    <span class="fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="tutorial_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="tutorial_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><link href="tutorial_files/vis-4.20.1/vis.css" rel="stylesheet">
<script src="tutorial_files/vis-4.20.1/vis.min.js"></script><script src="tutorial_files/visNetwork-binding-2.0.9/visNetwork.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>COSMOS-tutorial</h1>
                        <h4 class="author">A. Dugourd, A. Gabor and K. Zirngibl</h4>
            
            <h4 class="date">11/10/2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/COSMOSR/blob/master/vignettes/tutorial.Rmd"><code>vignettes/tutorial.Rmd</code></a></small>
      <div class="hidden name"><code>tutorial.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install from bioconductor</span>
<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"cosmosR"</span><span class="op">)</span>

<span class="co"># install the development version from GitHub</span>
<span class="co"># install.packages("remotes")</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"saezlab/cosmosR"</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>COSMOS (Causal Oriented Search of Multi-Omic Space) is a method that integrates phosphoproteomics, transcriptomics, and metabolomics data sets. COSMOS leverages extensive prior knowledge of signaling pathways, metabolic networks, and gene regulation with computational methods to estimate activities of transcription factors and kinases as well as network-level causal reasoning. This pipeline can provide mechanistic explanations for experimental observations across multiple omic data sets.</p>
<div class="figure">
<img src="../man/figures/intro_data.png" alt=""><p class="caption">data_intro_figure</p>
</div>
<p>First, we load the package</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/saezlab/COSMOSR">cosmosR</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="tutorial-section-signaling-to-metabolism" class="section level2">
<h2 class="hasAnchor">
<a href="#tutorial-section-signaling-to-metabolism" class="anchor"></a>Tutorial section: signaling to metabolism</h2>
<p>In this part, we can set up the options for the CARNIVAL run, such as timelimit and min gap tolerance.</p>
<p>The user should provide a path to its CPLEX/cbc executable</p>
<p>You can check the CARNIVAL_options variable to see all possible options that can be adjusted</p>
<p>In this example, we will use the built-in solver lpSolve. User should be aware that lpSolve should ONLY be used for TESTS. To obtain meaningful results, best solver is cplex, or cbc if not possible.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">CARNIVAL_options</span> <span class="op">&lt;-</span> <span class="fu">cosmosR</span><span class="fu">::</span><span class="fu"><a href="../reference/default_CARNIVAL_options.html">default_CARNIVAL_options</a></span><span class="op">(</span><span class="op">)</span>
<span class="co"># CARNIVAL_options$solverPath &lt;- "~/Documents/cplex"</span>
<span class="co"># CARNIVAL_options$solver &lt;- "cplex" #or cbc</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">solver</span> <span class="op">&lt;-</span> <span class="st">"lpSolve"</span> <span class="co">#or cbc</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">timelimit</span> <span class="op">&lt;-</span> <span class="fl">3600</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">mipGAP</span> <span class="op">&lt;-</span> <span class="fl">0.05</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">threads</span> <span class="op">&lt;-</span> <span class="fl">2</span></code></pre></div>
<p>In the next section, we prepare the input to run cosmosR The signaling inputs are the result of footprint based TF and kinase activity estiamtion For more info on TF activity estiamtion from transcriptomic data, see:<a href="https://github.com/saezlab/transcriptutorial" class="uri">https://github.com/saezlab/transcriptutorial</a> (Especially chapter 4)</p>
<p>Here we use of toy PKN, to see the full meta PKN, you can load it with data(meta_network).</p>
<p>The metabolites in the prior knowledge network are identified as XMetab_<em>PUBCHEMid<strong><em>compartment</em></strong></em> or XMetab_<em>BIGGid<strong><em>compartment</em></strong></em> or example “XMetab_<em>6804<strong><em>m</em></strong></em>”. The compartment code is the BIGG model standard (r, c, e, x, m, l, n, g). Thus we will first need to map whatever identifer for metabolite the data has to the one of the network. Genes are identified as XENTREZid (in the signaling part of network) or XGene####__ENTREZid (in the reaction network part of network)</p>
<p>The maximum network depth will define the maximum number of step downstream of kinase/TF COSMOS will look for deregulated metabolites. Good first guess for max depth could be around 6 (here is 15 for the toy dataset)</p>
<p>The differential experession data is used to filter out wrong TF-target interactions in this context after a pre-optimisation.</p>
<p>The list of genes in the differential expression data will also be used as a reference to define which genes are expressed or not (all genes in the diff_expression_data are considered expressed, and genes that are no in diff_expression_data are removed from the network)</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">toy_network</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">toy_signaling_input</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">toy_metabolic_input</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">toy_RNA</span><span class="op">)</span>
<span class="va">test_for</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess_COSMOS_signaling_to_metabolism.html">preprocess_COSMOS_signaling_to_metabolism</a></span><span class="op">(</span>meta_network <span class="op">=</span> <span class="va">toy_network</span>,
                                        signaling_data <span class="op">=</span> <span class="va">toy_signaling_input</span>,
                                        metabolic_data <span class="op">=</span> <span class="va">toy_metabolic_input</span>,
                                                      diff_expression_data <span class="op">=</span> <span class="va">toy_RNA</span>,
                                                      maximum_network_depth <span class="op">=</span> <span class="fl">15</span>,
                                                      remove_unexpressed_nodes <span class="op">=</span> <span class="cn">TRUE</span>,
                                                      CARNIVAL_options <span class="op">=</span> <span class="va">CARNIVAL_options</span>
                                                      <span class="op">)</span></code></pre></div>
<pre><code>## [1] "COSMOS: all 2 signaling nodes from data were found in the meta PKN"
## [1] "COSMOS: all 3 metabolic nodes from data were found in the meta PKN"
## [1] "COSMOS: 4653 of the 15919 genes in expression data were found as transcription factor target"
## [1] "COSMOS: 4653 of the 5303 transcription factor targets were found in expression data"
## [1] "COSMOS: removing unexpressed nodes from PKN..."
## [1] "COSMOS: 0 interactions removed"
## [1] "COSMOS: removing nodes that are not reachable from inputs within 15 steps"
## [1] "COSMOS: 10 from  19 interactions are removed from the PKN"
## [1] "COSMOS: 1 input/measured nodes are not in PKN any more: XMetab__190___c____ and 0 more."
## [1] "COSMOS: removing nodes that are not observable by measurements within 15 steps"
## [1] "COSMOS: 0 from  9 interactions are removed from the PKN"
## [1] "COSMOS:  1 interactions are removed from the PKN based on consistency check between TF activity and gene expression"
## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."
## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."
## [1] "COSMOS:  0 interactions are removed from the PKN based on consistency check between TF activity and gene expression"
## [1] "COSMOS: all 2 metabolic nodes from data were found in the meta PKN"
## [1] "COSMOS: 4653 of the 15919 genes in expression data were found as transcription factor target"
## [1] "COSMOS: 4653 of the 5303 transcription factor targets were found in expression data"</code></pre>
<p>In this part, we can set up the options for the actual run, such as timelimit and min gap tolerance.</p>
<p>The running time should be much higher here than in pre-optimisation. You cna increase the number of threads to use if you have many available CPUs.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">timelimit</span> <span class="op">&lt;-</span> <span class="fl">14400</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">mipGAP</span> <span class="op">&lt;-</span> <span class="fl">0.05</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">threads</span> <span class="op">&lt;-</span> <span class="fl">2</span></code></pre></div>
<p>This is where cosmosR run.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">test_result_for</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_COSMOS_signaling_to_metabolism.html">run_COSMOS_signaling_to_metabolism</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">test_for</span>,
                                                      CARNIVAL_options <span class="op">=</span> <span class="va">CARNIVAL_options</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."
## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."</code></pre>
<p>Finally, we process the results of the first cosmosR run, to translate gene names and metabolites name.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">metabolite_to_pubchem</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">omnipath_ptm</span><span class="op">)</span>
<span class="va">test_result_for</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_COSMOS_res.html">format_COSMOS_res</a></span><span class="op">(</span><span class="va">test_result_for</span>,
                                     metab_mapping <span class="op">=</span> <span class="va">metabolite_to_pubchem</span>,
                     measured_nodes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">toy_metabolic_input</span><span class="op">)</span>,
                                               <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">toy_signaling_input</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
                                     omnipath_ptm <span class="op">=</span> <span class="va">omnipath_ptm</span><span class="op">)</span></code></pre></div>
</div>
<div id="tutorial-section-metabolism-to-signaling" class="section level2">
<h2 class="hasAnchor">
<a href="#tutorial-section-metabolism-to-signaling" class="anchor"></a>Tutorial section: metabolism to signaling</h2>
<p>Before we run the metabolism to signaling part, we need to prepare again the inputs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">timelimit</span> <span class="op">&lt;-</span> <span class="fl">3600</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">mipGAP</span> <span class="op">&lt;-</span> <span class="fl">0.05</span>
<span class="va">CARNIVAL_options</span><span class="op">$</span><span class="va">threads</span> <span class="op">&lt;-</span> <span class="fl">2</span></code></pre></div>
<p>Now that the correct time is set up for the pre-optimisation run, we can prepare the inputs.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">test_back</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess_COSMOS_metabolism_to_signaling.html">preprocess_COSMOS_metabolism_to_signaling</a></span><span class="op">(</span>meta_network <span class="op">=</span> <span class="va">toy_network</span>,
                                        signaling_data <span class="op">=</span> <span class="va">toy_signaling_input</span>,
                                        metabolic_data <span class="op">=</span> <span class="va">toy_metabolic_input</span>,
                                                       diff_expression_data <span class="op">=</span> <span class="va">toy_RNA</span>,
                                                       maximum_network_depth <span class="op">=</span> <span class="fl">15</span>,
                                                       remove_unexpressed_nodes <span class="op">=</span> <span class="cn">FALSE</span>,
                                                       CARNIVAL_options <span class="op">=</span> <span class="va">CARNIVAL_options</span>
                                                       
<span class="op">)</span></code></pre></div>
<pre><code>## [1] "COSMOS: all 2 signaling nodes from data were found in the meta PKN"
## [1] "COSMOS: all 3 metabolic nodes from data were found in the meta PKN"
## [1] "COSMOS: 4653 of the 15919 genes in expression data were found as transcription factor target"
## [1] "COSMOS: 4653 of the 5303 transcription factor targets were found in expression data"
## [1] "COSMOS: removing nodes that are not reachable from inputs within 15 steps"
## [1] "COSMOS: 1 from  19 interactions are removed from the PKN"
## [1] "COSMOS: 1 input/measured nodes are not in PKN any more: X2305 and 0 more."
## [1] "COSMOS: removing nodes that are not observable by measurements within 15 steps"
## [1] "COSMOS: 11 from  18 interactions are removed from the PKN"
## [1] "COSMOS: 2 input/measured nodes are not in PKN any more: XMetab__124886___c____, XMetab__6426851___c____ and 0 more."
## [1] "COSMOS:  0 interactions are removed from the PKN based on consistency check between TF activity and gene expression"
## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."
## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."
## [1] "COSMOS:  0 interactions are removed from the PKN based on consistency check between TF activity and gene expression"
## [1] "COSMOS: all 1 signaling nodes from data were found in the meta PKN"
## [1] "COSMOS: all 1 metabolic nodes from data were found in the meta PKN"
## [1] "COSMOS: 4653 of the 15919 genes in expression data were found as transcription factor target"
## [1] "COSMOS: 4653 of the 5303 transcription factor targets were found in expression data"</code></pre>
<p>Then we can run cosmosR to connect metabolism to signaling. The running time here usually needs to be longer, as this problem seems to be harder to solve for CPLEX.</p>
<pre><code>## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."
## [1] "lpSolve does not scale well with large PKNs. This solver is mainly for testing purposes. To run COSMSO, we recommend using cplex, or cbc solvers."</code></pre>
<p>Finally we can format the result of the backward run as well (same as for forward run)</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">test_result_back</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_COSMOS_res.html">format_COSMOS_res</a></span><span class="op">(</span><span class="va">test_result_back</span>,
                                      metab_mapping <span class="op">=</span> <span class="va">metabolite_to_pubchem</span>,
                      measured_nodes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">toy_metabolic_input</span><span class="op">)</span>,
                                                <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">toy_signaling_input</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
                                      omnipath_ptm <span class="op">=</span> <span class="va">omnipath_ptm</span><span class="op">)</span></code></pre></div>
</div>
<div id="tutorial-section-merge-forward-and-backward-networks-and-visualise-network" class="section level2">
<h2 class="hasAnchor">
<a href="#tutorial-section-merge-forward-and-backward-networks-and-visualise-network" class="anchor"></a>Tutorial section: Merge forward and backward networks and visualise network</h2>
<p>Here we simply take the union of forward and backward runs to create a full network solution lopping between signaling, gene-regulation and metabolism. Since there is an overlapp between the result network of forward and backward run, you may optionally want to check if there are any node sign that are incoherent in the overlapp between the two solutions.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">full_sif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">test_result_for</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">test_result_back</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="va">full_attributes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">test_result_for</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="va">test_result_back</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>

<span class="va">full_sif</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">full_sif</span><span class="op">)</span>
<span class="va">full_attributes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">full_attributes</span><span class="op">)</span></code></pre></div>
<p>This function will generate a dynamic network plot centered on a given node of the network solution, and connecting it to measured nodes in the given range (here 5 steps).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">network_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/display_node_neighboorhood.html">display_node_neighboorhood</a></span><span class="op">(</span>central_node <span class="op">=</span> <span class="st">'BCAT1'</span>, 
                                           sif <span class="op">=</span> <span class="va">full_sif</span>, 
                                           att <span class="op">=</span> <span class="va">full_attributes</span>, 
                                           n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>

<span class="va">network_plot</span></code></pre></div>
<div id="htmlwidget-37f2946568e283b45d16" style="width:1600px;height:1600px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-37f2946568e283b45d16">{"x":{"nodes":{"id":["BCAT1","Enzyme3003__GGT6_GGTLC2_reverse","Enzyme7298__BCAT1","FOXM1","Metab__124886___c____","Metab__33032___c____","MYC"],"ZeroAct":[0,0,0,0,0,0,0],"UpAct":[1,1,1,1,1,1,1],"DownAct":[0,0,0,0,0,0,0],"AvgAct":[1,1,1,1,1,1,1],"NodeType":["","","","S","T","",""],"measured":[0,0,0,1,1,0,0],"type":["metab_enzyme","metab_enzyme","metab_enzyme","TF","metabolite","metabolite","TF"],"Activity":[1,1,1,1,1,1,1],"label":["BCAT1","Enzyme3003__GGT6_GGTLC2_reverse","Enzyme7298__BCAT1","FOXM1","Metab__124886___c____","Metab__33032___c____","MYC"],"color":["green","green","green","green","green","green","green"],"shape":["square","square","square","diamond","dot","dot","diamond"],"shadow":[false,false,false,true,true,false,false]},"edges":{"from":["MYC","FOXM1","BCAT1","Enzyme3003__GGT6_GGTLC2_reverse","Metab__33032___c____","Enzyme7298__BCAT1"],"to":["BCAT1","MYC","Enzyme7298__BCAT1","Metab__124886___c____","Enzyme3003__GGT6_GGTLC2_reverse","Metab__33032___c____"],"sign":[1,1,1,1,1,1],"weigth":[1,1,1,1,1,1],"color":["grey","grey","grey","grey","grey","grey"],"arrows.to.type":["arrow","arrow","arrow","arrow","arrow","arrow"],"enabled":[true,true,true,true,true,true],"scaleFactor":[1,1,1,1,1,1]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false}},"groups":null,"width":1600,"height":1600,"idselection":{"enabled":true,"style":"width: 200px; height: 26px;\n                                              background: #f8f8f8;\n                                              color: darkblue;\n                                              border:none;\n                                              outline:none;","useLabels":true,"main":"Select by id"},"byselection":{"enabled":false,"style":"width: 150px; height: 26px","multiple":false,"hideColor":"rgba(200,200,200,0.5)","highlight":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","highlight":{"enabled":true,"hoverNearest":false,"degree":1,"algorithm":"all","hideColor":"rgba(200,200,200,0.5)","labelOnly":true},"collapse":{"enabled":false,"fit":false,"resetHighlight":true,"clusterOptions":null,"keepCoord":true,"labelSuffix":"(cluster)"}},"evals":[],"jsHooks":[]}</script><p>This network represent the flow of activities that can connect FOXM1 up-regulation with glutathione (CID 124886) accumulation. Here, FOXM1 can activate MYC, which in turn activate BCAT1. The activation of BCAT1 can lead to the increased production of glutamate (CID 33032), whioch in turn can be converted to glutathione GGT enzymes.</p>
<p>It is important to understand that each of this links are hypothetical. The come from a larger pool of potential molecular interactions present in multiple online databases and compiled in omnipath, STITCH and recon metabolic network. They exist in the literature and are interactions that are known to potentially exists in other experimental contexts. Thus, COSMOS compile all those potential interactions together and proposes a coherent set that can explain the data at hand. Here, such a set of mechanistic hypothesis is plotted as a network connecting FOXM1 and glutathione production.</p>
<p>Those links should however be considered only as potential mechanistic connections, and will need to be further confirmed experimentally.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] cosmosR_0.99.2
## 
## loaded via a namespace (and not attached):
##   [1] segmented_1.3-4      Category_2.56.0      bitops_1.0-7        
##   [4] fs_1.5.0             bit64_4.0.5          doParallel_1.0.16   
##   [7] httr_1.4.2           rprojroot_2.0.2      tools_4.0.2         
##  [10] bslib_0.2.4          utf8_1.2.1           R6_2.5.0            
##  [13] KernSmooth_2.23-20   DBI_1.1.1            BiocGenerics_0.36.1 
##  [16] colorspace_2.0-1     tidyselect_1.1.1     curl_4.3.1          
##  [19] bit_4.0.4            compiler_4.0.2       cli_2.5.0           
##  [22] textshaping_0.3.3    graph_1.68.0         Biobase_2.50.0      
##  [25] CARNIVAL_1.2.0       desc_1.3.0           sass_0.3.1          
##  [28] scales_1.1.1         readr_1.4.0          genefilter_1.72.1   
##  [31] proxy_0.4-25         RBGL_1.66.0          rappdirs_0.3.3      
##  [34] pkgdown_1.6.1        systemfonts_1.0.1    stringr_1.4.0       
##  [37] digest_0.6.27        mixtools_1.2.0       rmarkdown_2.8       
##  [40] pkgconfig_2.0.3      htmltools_0.5.1.1    bcellViper_1.26.0   
##  [43] dbplyr_2.1.1         fastmap_1.1.0        htmlwidgets_1.5.3   
##  [46] rlang_0.4.11         rstudioapi_0.13      RSQLite_2.2.7       
##  [49] visNetwork_2.0.9     jquerylib_0.1.4      generics_0.1.0      
##  [52] jsonlite_1.7.2       viper_1.24.0         dplyr_1.0.6         
##  [55] RCurl_1.98-1.3       magrittr_2.0.1       Matrix_1.3-3        
##  [58] Rcpp_1.0.6           munsell_0.5.0        S4Vectors_0.28.1    
##  [61] fansi_0.4.2          lifecycle_1.0.0      stringi_1.5.3       
##  [64] yaml_2.2.1           UniProt.ws_2.30.0    MASS_7.3-54         
##  [67] BiocFileCache_1.14.0 org.Hs.eg.db_3.12.0  grid_4.0.2          
##  [70] blob_1.2.1           parallel_4.0.2       crayon_1.4.1        
##  [73] lattice_0.20-44      splines_4.0.2        annotate_1.68.0     
##  [76] hms_1.0.0            knitr_1.33           dorothea_1.2.2      
##  [79] pillar_1.6.0         igraph_1.2.6         lpSolve_5.6.15      
##  [82] codetools_0.2-18     stats4_4.0.2         XML_3.99-0.6        
##  [85] glue_1.4.2           evaluate_0.14        vctrs_0.3.8         
##  [88] foreach_1.5.1        gtable_0.3.0         purrr_0.3.4         
##  [91] kernlab_0.9-29       assertthat_0.2.1     cachem_1.0.4        
##  [94] ggplot2_3.3.3        xfun_0.22            xtable_1.8-4        
##  [97] e1071_1.7-6          ragg_1.1.2           class_7.3-19        
## [100] survival_3.2-11      tibble_3.1.1         iterators_1.0.13    
## [103] AnnotationDbi_1.52.0 memoise_2.0.0        IRanges_2.24.1      
## [106] ellipsis_0.3.2       GSEABase_1.52.1</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Aurélien Dugourd, Attila Gabor, Katharina Zirngibl.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
